lib/jekyll/filters.rb

```diff
@@ -389,7 +389,11 @@ def groupable?(element)
     private
     def item_property(item, property)
       if item.respond_to?(:to_liquid)
-        item.to_liquid[property.to_s]
+        res = item.to_liquid
+        property.to_s.split(".").each do |attr|
+          res = res[attr]
+        end
+        res
       elsif item.respond_to?(:data)
         item.data[property.to_s]
```

Comment:
That's great! Could you add a similar test to our test-suite as well? Thanks :slightly_smiling_face: